\documentclass{article} % For LaTeX2e
\usepackage{nips15submit_e,times}
\usepackage{hyperref}
\usepackage{url}
\usepackage{graphicx}
\usepackage{bbm}
\usepackage{amsmath}
\usepackage[psamsfonts]{amssymb}
\graphicspath{ {images/} }
\usepackage{algorithm}
\usepackage{algorithmic}
\usepackage[square,numbers]{natbib}

\bibliographystyle{unsrtnat}

%\documentstyle[nips14submit_09,times,art10]{article} % For LaTeX 2.09
\title{Multi-region demixing of brain transcriptome}
% \addbibresource{nmf_multi.bib}

\author{
Lior Kirsch \\ Bar Ilan university \\ Ramat Gan 52900 Israel \\ 
\And
Gal Chechik \\ Bar Ilan university \\ Ramat Gan 52900 Israel \\
\texttt{gal.chechik@biu.ac.il} \\
}
\newcommand{\fix}{\marginpar{FIX}}
\newcommand{\new}{\marginpar{NEW}}
\newcommand{\reals}{\mathbb{R}}
\newcommand{\W}{W}
\renewcommand{\eqref}[1]{Eq.~(\ref{#1})}
\newcommand{\figref}[1]{Fig.~(\ref{#1})}
\newcommand{\figureref}[1]{Figure~(\ref{#1})}
%\nipsfinalcopy % Uncomment for camera-ready version
\begin{document}
\maketitle
\begin{abstract}
    High quality and large scale measurements of gene expression in the brain have been collected across many species, many regions and many ages, but unfortunately, almost all these measurements reflect  expression of an uncontrolled mix from various cell types including neurons, glia and blood cells. This fact hinders the insights that can be gained by analyzing datasets of gene expression in the brain. Inferring cell-type specific information from these mixtures is a key challenge. We propose here a probabilistic approach to blindly demix brain transcriptome samples, building on variability of cell-type proportions across samples. We further describe a model that takes into account that tissues in different brain structures differ in their expression profiles by introducing soft-sharing between cell-type-specific profiles from different brain regions. We describe an efficient algorithm to estimate the parameters of the model, as well as techniques to determine some of its hyper parameters from prior knowledge. We study the regimes where multi-region demixing works well using controlled data and evaluate the approach using RNA-seq profiles measured from multiple regions in postmortem human brains. Predictions are validated against ground-truth data of cell-type specific profiles measured in corresponding regions in the mouse brain, showing that soft demixing successfully outperforms baselines of demixing unified and individual regions. 
\end{abstract}

\section{Introduction}
\label{introduction}
Recent advances in molecular biology techniques allow to measure transcriptome profiles across the brain at unprecedented quality and magnitude. Large datasets of gene expression in the brain are now available, covering  many brain regions, ages and species. RNA sequencing techniques now provides accurate genome-wide counts of transcripts for each gene and splice variant. Unfortunately however, when gene expression is measured in a complex tissue, like the brain, the measured transcripts reflect an uncontrolled mix of expression from multiple cell types including neurons, astrocytes, oligodendrocytes and even blood and immune system cells. This limits the analysis and understanding that can be gained from gene expression measurements of the brain. Although several techniques have been developed to isolate and purify cells of individual types and measure their transcription profiles \cite{okaty2011cell,barres2014,darmanis2015survey}, these techniques are costly and complex, and only a handful of cell-type isolated profiles have been collected so far. 

The current paper proposes a computational technique to demix gene expression measurements in the brain. The idea is simple: when multiple samples are collected from the brain, their cellular composition often varies slightly from one sample to another. Each sample therefore reflects mixtures with different proportions of cell types. We propose here to model transcriptome profiles as a mixture of contributions from different cell types, and describe how the latent component that are shared across samples can be found.

Developing robust methods to extract cell-type specific signals from transcriptome mixtures, has a potential to be extremely useful. It could provide deeper analysis of data that was already collected and is very hard and costly to reproduce. For example, collecting brain banks in human patients with brain disorders is a complex and lengthy process, so extracting cell-type level information from existing brain transcript datasets can help delineating the role of neurons and glia in diseased like Alzheimer (AD) or Epilepsy.

In the mouse brain, a demixing approach was recently applied to ISH transcriptome measures \cite{grange2014cell}, where expression profiles were demixed using known cell-type-specific profiles. Here we take a further step and apply blind demixing, inferring both the mixture components and their proportions. We use isolated cell-type profiles for validation but not during training. 

%Demixing transcriptome profiles into their components has been applied in various domains outside neuroscience \cite{}.

Importantly, collections of transcriptome measurements in the brain have a unique structure that we suggest to benefit from. In the brain, gene expression profile changes from one brain region to the other, in a way that often follows the developmental hierarchy of brain regions \cite{zapala2005}. This suggests that we can use a-priori knowledge about regions which are expected to share similar expression profiles. For instance, pyramidal cells can be found in most neocortex areas, hence may express a highly preserved set of genes. At the same time, neurons in the hippocampus (an archeo-cortical region) may exhibit an expression profile that is less similar to the neocortical profile. Figure 1 shows a hierarchy of brain regions based on embryonic development developed by the Allen brain institute. 

\begin{figure}[!hbt]
  \begin{minipage}[c]{0.65\textwidth}
    \includegraphics[width=\textwidth]{tree}
  \end{minipage}\hfill
  \begin{minipage}[c]{0.3\textwidth}
    \caption{Hierarchical relations among brain regions. An ontology of regions of the human brain has been created based on neural development. Colored boxes correspond to brain regions that we analyzed below. The color scheme is based on caudal-rostral axis of brain regions, from red hindbrain to blue forebrain.} 
    \label{fig:bro}
  \end{minipage}
\end{figure}

This hierarchical structure suggests that when applying demixing to brain-omics data, one should be careful not to treat all brain samples as homogeneously containing the same set of neurons and underlying expressed genes. Instead, we can take into account the relative similarity among brain structures. 

To combine the various sources of prior information, our approach is based on a probabilistic model with three components. First, a set of samples from a single brain region is represented as a mixture of a smaller number of components. Second, we can include a prior on each of these underlying cell-type-specific profiles based on previous measurements. Third, we add pairwise attractive potentials, pushing expression profiles of related regions to be similar.
 
This paper makes three main novel contributions. First, it describes for the first time blind demixing and multi-region demixing of transcriptome measurements. Second, it characterizes the regime of parameters where blind demixing works best in controlled mixtures. Finally, it demonstrates that demixing can uncover latent factors in RNA-seq measurements from multiple human brain regions.
 
The paper is organized as follows: We start by describing a probabilistic model of expression mix in a single region and and multiple regions. We then describe algorithms to optimize the parameters of the these model. Section 5, evaluate the regime of parameters where the demixing algorithm works well using we a controlled mix dataset that we created. Section 6 evaluates the algorithms in expression dataset from postmortem human brains.

% ==========================================
\section{A probabilistic model for multi-region demixing}
\input{method}

% ==========================================
\section{Related methods}
Demixing of transcriptome measurements into cell-type specific profiles has been mostly applied when a set of ground-truth underlying factors is known. \citet{shen2010cell} described demixing of 
whole-blood gene expression datasets from kidney transplants. \citet{grange2014cell} used isolated profiles of neurons and glia \cite{okaty2011cell} to map cell-type proportions across ISH measures of the mouse brain. Cell type demixing is also widely used in the context of immunology studies \cite{shen2013computational}.

Joint non negative matrix factorization of several matrices has been previously studied by \citet{lee2009group}. Their formulation introduced  both attractive and repulsive pairwise potentials, optimized using multiplicative updates. A Bayesian group factorization model was presented in \citet{shin2012bayesian}. Somewhat related, is the work by \citet{wang2012group}, who studied matrix factorization where some factors are shared across matrices.


\input{experiments}



% ==========================================
\section{Summary and future directions}
We presented a probabilistic model for multi-region demixing approach aiming to estimate cell-type specific expression profiles from expression mixtures. We provide a simple and efficient way to solve it using repeated ANLS steps. We find that soft-sharing improves the accuracy of reconstruction as compared to demixing of individual regions, demixing of unified brain and random samples. This was consistent in both controlled samples where the ground truth was known and in postmortem human brains were we validated against mouse cell-type specific profiles.

The results shown above are expect to improve dramatically as further prior information is incorporated into the model. Some of that information is already available \cite{darmanis2015survey}, and we will add it in the near future.  Specifically, these include recent new measures of cell-type specific transcriptome from human cells, which can be used to determine the expected profiles $\mu_k$. Furthermore, estimates of coordinated expression among gene pairs can be used to determine the co-variation parameters in the model $\Sigma$.
As another direction, we expect that profiling might benefit from selecting subsets of genes that are strongly associated with specific cell types, and filtering genes that are not differentially expressed.

\newpage
\bibliography{nmf_multi}

\end{document}

